src/app/App.js

```diff
@@ -287,6 +287,10 @@ class App extends EventEmitter {
 			console.log('Link clicked outside app\'s base path');
 			return false;
 		}
+		// Prevents navigation if it's a hash change on the same url.
+		if (uri.getHash() && utils.isCurrentBrowserPath(path)) {
+			return false;
+		}
```

Comment:
Just so I understand it... this isn't really changing any logic, right? Just moving the condition where it actually makes sense?